developerportal/apps/common/models.py

```diff
@@ -26,5 +28,14 @@ class BasePage(Page):
         related_name="+",
     )
 
+    #  A single place to store/define cache keys that need invalidation
+    # when the object is saved
+    _bulk_invalidation_cache_keys = []
+
     class Meta:
         abstract = True
+
+    @transaction.atomic()
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        cache.delete_many(self._bulk_invalidation_cache_keys)
```

Comment:
how about we only delete the cache key for the currently saved record, i.e. use `self.pk` in the cache key?

Reply:
Hmm - maybe, yeah. I think the next piece of work re cacheing (which will broaden out how we do it across models and will inevitably have some tuning involved) might be a better place to explore that. Right now, `Topics._bulk_invalidation_cache_keys` is a list with just one member, so there's not much to gain in efficiency/specificity terms.

Could you live with this as-is for now?

I quite like the idea of using `self.pk` to as a label/tag for a set of keys to do bulk invalidation on (via something like https://gist.github.com/stevejalim/a55d53994a455dd3e95e)

Reply:
Fair! :+1: I'm still figuring out how to balance what's currently needed from the caching thing with what will be needed which I probably don't know enough about. So overall I do think it's fine for now and I hope I still remember all those thoughts when I review the next cache PR.